File: frontend/src/types/interview.ts
export interface Role {
  id: string;
  title: string;
  description: string;
}

export interface QuestionAnswer {
  id: string;
  order_index: number;
  question_text: string;
  answer_text: string | null;
}

export interface InterviewSession {
  id: string;
  candidate_id: string;
  resume_id: string;
  role_id: string;
  status: string;
  created_at: string;
  questions: QuestionAnswer[];
}

export interface NextQuestionResponse {
  session_id: string;
  question: QuestionAnswer | null;
  is_complete: boolean;
}

export interface InterviewSummary {
  id: string;
  session_id: string;
  overall_score: number | null;
  strengths: string | null;
  weaknesses: string | null;
  summary_text: string;
  created_at: string;
}

File: backend/app/schemas/interview.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class InterviewStartRequest(BaseModel):
    resume_id: str
    role_id: str


class QuestionOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    order_index: int
    question_text: str
    answer_text: str | None = None


class InterviewSessionOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    candidate_id: str
    resume_id: str
    role_id: str
    status: str
    created_at: datetime
    questions: list[QuestionOut] = []


class AnswerSubmitRequest(BaseModel):
    question_id: str
    answer_text: str


class NextQuestionResponse(BaseModel):
    session_id: str
    question: QuestionOut | None
    is_complete: bool
